### PR TITLE
官网首页的搜索框，当按下键盘s的时候，会获取焦点

### PR DESCRIPTION
**Pull request by Manweill**
First of all, thank you for your contribution! :-)

Please makes sure that these checkboxes are checked before submitting your PR, thank you!

* [x] Make sure that you propose PR to right branch: bugfix for `master`, feature for latest active branch `feature-x.x`.
* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

Extra checklist:

**if** *isBugFix* **:**

  * [x] Make sure that you add at least one unit test for the bug which you had fixed.

**elif** *isNewFeature* **:**

  * [x] Update API docs for the component.
  * [x] Update/Add demo to demonstrate new feature.
  * [x] Update TypeScript definition for the component.
  * [x] Add unit tests for the feature.

### CODE REVIEW

**Comment by Manweill** [comment]
@benjycui

File: site/theme/template/Layout/Header.jsx [hunk 1]
```diff
@@ -36,6 +37,12 @@ export default class Header extends React.Component {
           this.setState({ menuMode: 'horizontal' });
         },
       });
+    document.addEventListener('keyup', (event) => {
+      const { focused } = this.state;
+      if (event.keyCode === 83 && !focused) {
```
Comment (benjycui) [inline site/theme/template/Layout/Header.jsx]: 其实 `!focused` 这个判断是不需要的吧，反复调用 `.focus()` 会有什么问题么？

Comment (Manweill) [inline site/theme/template/Layout/Header.jsx]: 应该没有影响，应该是上个版本遗留代码，忘记复查keyup这块~

**Comment by benjycui** [comment]
突然想起一个问题，在 https://ant.design/components/form/#components-form-demo-horizontal-login 的 demo 中输入 s ，也会导致跳到 header 并 focus ，所以跳转之前应该判断一下 target。

**Comment by benjycui** [comment]
需要判断 e.target ，如果是 input 或者 textarea 就不 focus。

后面如果碰到其他元素也有类似的问题，在补充吧。

**Comment by afc163** [comment]
@Manweill

**Comment by paranoidjk** [comment]
Nice job, mark,  wating to copy this for antd-mobile.

**Comment by Manweill** [comment]
增加了e.target === document.body的判断

**Pull request merged**